nmdc_automation/import_automation/activity_mapper.py

```diff
@@ -200,39 +228,45 @@ def workflow_execution_mapper(self) -> None:
                 workflow["Type"]
             )
-            # quick fix because nmdc-schema does not support [], even though raw product has none
-            if len(has_output_list) == 0:
-                logging.warning("No outputs.  That seems odd.")
-                has_output_list = ["None"]
-            # input may be none for metagenome sequencing
-            if len(has_inputs_list) == 0:
-                has_inputs_list = ["None"]
-            # Lookup the nmdc database class
-            database_workflow_execution_set = getattr(self.nmdc_db, workflow["Collection"])
-            # Lookup the nmdc schema range class
-            database_workflow_execution_range = getattr(nmdc, workflow["WorkflowExecutionRange"])
-            # Mint an ID
-            workflow_execution_id = self.get_workflow_execution_id(workflow["Type"])
-            database_workflow_execution_set.append(
-                database_workflow_execution_range(
-                    id=workflow_execution_id,
-                    name=workflow["Workflow_Execution"]["name"].replace("{id}", workflow_execution_id),
-                    git_url=workflow["Git_repo"],
-                    version=workflow["Version"],
-                    execution_resource=self.import_data["Workflow Metadata"][
-                        "Execution Resource"
-                    ],
-                    started_at_time=datetime.datetime.now(pytz.utc).isoformat(),
-                    has_input=has_inputs_list,
-                    has_output=has_output_list,
-                    type=workflow["Type"],
-                    ended_at_time=datetime.datetime.now(pytz.utc).isoformat(),
-                    was_informed_by=self.nucelotide_sequencing_id,
+            # We can't make a valid workflow execution without inputs and outputs
+            if not has_inputs_list or not has_output_list:
+                logging.error(
+                    f"Skipping {workflow['Name']} due to missing inputs or outputs"
                 )
-            )
+                continue
+            # Mint an ID if needed
+            wf_id = self.workflow_execution_ids.get(workflow["Type"], None)
+            if wf_id is None:
+                # mint an ID
+                wf_id = self.runtime.minter(workflow["Type"]) + "." + self.iteration
+                # store the ID
+                self.workflow_execution_ids[workflow["Type"]] = wf_id
+
+
+            # Create the workflow execution object
+            record = {
+                "id": wf_id,
+                "name": workflow["Workflow_Execution"]["name"].replace("{id}", wf_id),
+                "type": workflow["Type"],
+                "has_input": has_inputs_list,
+                "has_output": has_output_list,
+                "git_url": workflow["Git_repo"],
+                "version": workflow["Version"],
+                "execution_resource": self.import_data["Workflow Metadata"]["Execution Resource"],
```

Comment:
add [processing_institution](https://microbiomedata.github.io/nmdc-schema/processing_institution/), default value of JGI.

Reply:
disregard, we are overloading execution_resource. revisit at a later time.